### PR TITLE
[plug-in] [vscode] Add some missing entries in env object

### PR DESCRIPTION
**Pull request by benoitf**
It's addressing partially #4446 and makes env tests passing at https://github.com/theia-ide/theia/issues/4224

machineID is using mac address on nodejs while on workers it's a random ID.

- [x] CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19584

Change-Id: I589ede711a18322b1afffe0479dace781f450045
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

File: packages/plugin-ext/src/plugin/env.ts [hunk 1]
```diff
@@ -48,7 +56,32 @@ export class EnvExtImpl {
...
     getClientOperatingSystem(): Promise<theia.OperatingSystem> {
         return this.proxy.$getClientOperatingSystem();
     }
+
+    get appName() {
+        return EnvExtImpl.APP_NAME;
+    }
+
+    get appRoot() {
+        return __dirname;
```
Comment (mmorhun) [inline packages/plugin-ext/src/plugin/env.ts]: Will this work in web worker?

Comment (benoitf) [reply packages/plugin-ext/src/plugin/env.ts]: will check, I splitted class class for node-mac, I will see how to handle that
good catch

File: packages/plugin-ext/package.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
     "@theia/task": "^0.5.0",
     "@theia/workspace": "^0.5.0",
     "decompress": "^4.2.0",
+    "getmac": "^1.4.6",
```
Comment (akosyakov) [inline packages/plugin-ext/package.json]: @marcdumais-work Could you please use your tool to verify this dep?

File: packages/plugin-ext/src/plugin/env.ts (hunk 1 above)
Comment (benoitf) [reply packages/plugin-ext/src/plugin/env.ts]: @mmorhun : ok I've splitted in two classes, one for worker and one for nodejs

in worker env, we don't provide filesystem path as there is no FS BTW

File: packages/plugin-ext/package.json (hunk 2 above)
Comment (marcdumais-work) [reply packages/plugin-ext/package.json]: Florent and I did the analysis anyway for the new dependency and its own recursive dependency tree. All licenses found are Foundation-approved: 

Screenshot from 2019-04-24 10-29-59

**Pull request merged**